### PR TITLE
[ecobee] Updates in prep for Dec 1 changes to Ecobee authorization process

### PR DESCRIPTION
**Pull request by mhilbush**
On 1 December 2020, Ecobee will make changes to their authorization process. Users who authorized the binding prior to December 1 will continue to work using the old process, however, Ecobee recommends that users reauthorize using the new process. In addition, any new installations of the binding after 1 Dec will use the new process.

The changes included in this PR support the new process, as well as include several small fixes that will help ensure the reauthorization process runs smoothly.

Also related to #9039

A similar set of changes will be pushed for 2.5.x.

Signed-off-by: Mark Hilbush <mark@hilbush.com>

### CODE REVIEW

**Comment by mhilbush** [comment]
I added the M3 milestone tag. Given that the Ecobee back end changes take effect on 1 Dec, it's very desirable to get this into milestone 3.

File: bundles/org.openhab.binding.ecobee/README.md [hunk 1]
```diff
@@ -57,21 +57,43 @@ After you have installed the binding, you can create the Account thing using the
...
+Ecobee recommends that users who authorized with Ecobee prior to 1 Dec 2020 should reauthorize their application as the new process affords greater security.
+While the binding will continue to work using the old authorization method, it's recommended that you reauthorize the binding using the following process.
+
+- You may need to update openHAB to get the latest version of the binding
+
+- In the MyApps section of your Ecobee Portal, remove the application using the **Remove App** function.
+
+- Wait up to one hour for the binding to do a token refresh with the Ecobee servers.
+
+- At this point the Ecobee Account thing should be OFFLINE with a CONFIGURATION_PENING status.
```
Comment (fwolter) [inline bundles/org.openhab.binding.ecobee/README.md:87]:
```suggestion
- At this point the Ecobee Account thing should be OFFLINE with a CONFIGURATION_PENDING status.
```

**Review by fwolter — CHANGES_REQUESTED** [review]
I didn't see this in time. Next time you could add the critical label.

File: bundles/org.openhab.binding.ecobee/README.md (hunk 1 above)
Comment (mhilbush) [reply bundles/org.openhab.binding.ecobee/README.md]: Thanks for catching this.

**Review by fwolter — APPROVED** [review]
LGTM

**Comment by fwolter** [comment]
Shall I cherry-pick this to the 2.5.x branch?

**Comment by mhilbush** [comment]
I already submitted a separate PR for 2.5.x. It contains this change, as well as another fix.

**Pull request merged**